logstash-core-event-java/src/main/java/com/logstash/ConvertedList.java

```diff
@@ -0,0 +1,224 @@
+package com.logstash;
+
+import org.jruby.RubyArray;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+import static com.logstash.Valuefier.convert;
+
+public class ConvertedList<T> implements List<T>, Collection<T>, Iterable<T> {
+    private final List<T> delegate;
+
+    public ConvertedList(List<T> delegate) {
+        this.delegate = delegate;
+    }
+    public ConvertedList() {
+        this.delegate = new ArrayList<>();
+    }
+
+    public static List<Object> newFromList(List<Object> list) {
```

Comment:
I think it would make more sense to have the return object be a `ConvertedList` and not the `List` generic since this is a static factory that generates `ConvertedList` specifically, just like the class constructor.